lib/frameworks/jasmine.py

```diff
@@ -1,11 +1,81 @@
-import json, retrying, urllib
+# coding: utf-8
+
+import json, retrying, urllib, sys
 from selenium.webdriver.support.ui import WebDriverWait
 from itertools import groupby
 from xml.etree import ElementTree
 from lib import log
 
 webDriverWaitTimeout = 300
 
+from concurrent import futures
+
+def runTestsInParallel(drivers, timeout):
+    log.write("Calculating number of tests ... ")
+    tests = getTests( drivers[0]["driver"], drivers[0]["testsUrl"] )
+    log.writeln("%d tests found" % len( tests ))
+
+    testResults = runTestsInPoolDrivers(drivers, tests)
+
+    suites = groupTestSuites( map(lambda x: x["json"], testResults) )
+    xmlSuites = map(lambda x: ElementTree.tostring(x), groupXmlSuites( map(lambda x: x[ "junit" ], testResults) ))
+
+    return {
+        "json": {
+            "suites": suites,
+            "durationSec": sum( map(lambda x: x["durationSec"], suites) ),
+            "passed": all( map(lambda x: x["passed"], suites) )
+        },
+        "junit": '<?xml version="1.0" encoding="UTF-8" ?>\n<testsuites>\n%s\n</testsuites>\n' % "\n".join(xmlSuites)
+    }
+
+def runTestsInPoolDrivers(drivers, tests):
+    test_results = []
+    counter = 0
+
+    with futures.ThreadPoolExecutor(max_workers=len(drivers)) as executor:
+      executions = []
+      for test in tests:
+          counter += 1
+          executions.append( executor.submit(runTestByDriversPool, drivers, test, counter) )
+      for execution in executions:
+          exception = execution.exception()
+          if exception is not None:
+              log.writeln( "Got exception that broke everything: " + str( exception ) + "\n" )
+          else:
+              testResult = execution.result()
+              test_results.append( testResult )
+
+    return test_results
+    
+def runTestByDriversPool(driversPool, test, counter):
+    driver = None
+    try:
```

Comment:
This is overhead. You do not need any `try-except-finally` statements in this method.